app/src/main/java/org/mozilla/fenix/quickactionsheet/QuickActionSheet.kt

```diff
@@ -28,80 +33,66 @@ class QuickActionSheet @JvmOverloads constructor(
     attrs: AttributeSet? = null,
     defStyle: Int = 0,
     defStyleRes: Int = 0
-) : LinearLayout(context, attrs, defStyle, defStyleRes) {
+) : LinearLayout(context, attrs, defStyle, defStyleRes), CoroutineScope {
+
+    private val job: Job
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Main + job
+
+    private lateinit var handle: ImageButton
+    private lateinit var linearLayout: LinearLayout
+    private lateinit var quickActionSheetBehavior: QuickActionSheetBehavior
 
     init {
+        job = Job()
```

Comment:
This job should have some kind of lifecycle. I believe we want to cancel if the view is no longer onscreen and we don't want to do any work if we aren't onscreen yet. How about initializing in onAttachedToWindow and canceling in onDetachedFromWindow?

Reply:
Fixed this 😄